app/models/tag.rb

```diff
@@ -32,14 +32,20 @@ def self.from_csv(data, assignment_id)
     result
   end
 
-  def self.from_yml(data, assignment_id)
+  def self.from_yml(data, assignment_id, allow_ta_upload = false)
     admins = Hash[Admin.pluck(:user_name, :id)]
     begin
       tag_data = data.map do |row|
         row = row.symbolize_keys
-        name, description, user_id = row[:name], row[:description], admins[row[:user]]
+        author = admins[row[:user]]
+        # Allow TAs with proper permissions to upload yml tag data
```

Comment:
I see what you're doing here, but currently we don't allow TAs to create new tags (in our `TagPolicy` the permissions require being an instructor). While it's possible to change this to allow TAs to create tags (say, if the `manage_assessments` permission is enabled), it shouldn't happen in this PR. So for now, please revert these changes to this method, including the additional parameter.